src/main/java/updater/JarUpdater.java

```diff
@@ -0,0 +1,278 @@
+ *
+ * Replaces a JAR file and optionally start that JAR
+ *
+ * Options:
+ * --source string path of source JAR file to replace old JAR
+ * --target string path of target JAR file to be replaced
+ * --execute-jar "y" to execute target JAR after replacement
+ */
+public class JarUpdater extends Application {
+    private final ExecutorService pool = Executors.newSingleThreadExecutor();
+
+    private static final int MAX_RETRY = 10;
+    private static final int WAIT_TIME = 2000;
+    private static final String BACKUP_FILENAME_SUFFIX = "_backup";
+    private static final String ERROR_ON_UPDATING_MESSAGE =
+            "Update cannot be applied. Update will be aborted. Please close this window.";
+
+    private Label updatingLabel;
+
+
+    public static void main(String[] args) { // NOPMD
+        Application.launch(args);
+    }
+
+    @Override
+    public void start(Stage stage) {
+        initLogger();
+        showWaitingWindow(stage);
+        pool.execute(() -> {
+            try {
+                run();
+            } catch (IOException | IllegalArgumentException e) {
+                log(e.getMessage());
+            }
+
+            quit();
+        });
+    }
+
+    private void showWaitingWindow(Stage stage) {
+        stage.setTitle("Applying Updates");
+        VBox windowMainLayout = new VBox();
+        Group root = new Group();
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+        scene.setRoot(windowMainLayout);
+
+        updatingLabel = new Label();
+        updatingLabel.setText("Please wait. HubTurbo is being updated...");
+        updatingLabel.setPadding(new Insets(50));
+
+        windowMainLayout.getChildren().addAll(updatingLabel);
+
+        stage.show();
+    }
+
+    private void run() throws IllegalArgumentException, IOException {
+        HashMap<String, String> commandLineArgs = new HashMap<>(getParameters().getNamed());
+
+        String sourceJarPath = commandLineArgs.get("source");
+        String targetJarPath = commandLineArgs.get("target");
+        String executeJarOption = commandLineArgs.get("execute-jar");
+
+        if (sourceJarPath == null || targetJarPath == null) {
+            throw new IllegalArgumentException("Please specify source and target files.");
+        } else {
+            log("source: " + sourceJarPath);
+            log("target: " + targetJarPath);
+        }
+
+        File sourceJarFile = new File(sourceJarPath);
+        File targetJarFile = new File(targetJarPath);
+
+        if (!sourceJarFile.exists()) {
+            throw new IOException("Such source file does not exist, aborting.");
+        }
+
+        if (!prepareTargetJarFile(targetJarFile)) {
+            showCriticalErrorOnUpdating();
+            return;
+        }
+
+        log("Moving source to target");
+        if (!moveFile(sourceJarFile.toPath(), targetJarFile.toPath())) {
+            restoreBackup(targetJarFile);
+            showCriticalErrorOnUpdating();
+            return;
```

Comment:
I don't think we need the `return`s here, seeing as execution ends with these methods.